### ISSUE TITLE
Error when querying for optional rdfs:label

### ISSUE THREAD

**Issue by fanavarro** [ontop/ontop#190]
### Description

When I try to perform a query like the following:
```
PREFIX : <http://www.semanticweb.org/fabad/ontologies/2017/1/untitled-ontology-215#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

select distinct * where {
	?s a :Person .
	OPTIONAL { ?s rdfs:label ?label . }
}
```
I get the following error messaje:
```
it.unibz.inf.ontop.model.OBDAException: Error rewriting and unfolding into SQL
Impossible to get data definition for: http://www.w3.org/2000/01/rdf-schema#label(URI("http://www.semanticweb.org/fabad/ontologies/2017/1/untitled-ontology-215#person{}",t0_227f1),labelf5), type: null

###################################################
##    Debugging information (for the authors)    ##
###################################################

	at it.unibz.inf.ontop.owlrefplatform.owlapi.QuestOWLStatement.executeTuple(QuestOWLStatement.java:109)
	at it.unibz.inf.ontop.protege.views.QueryInterfaceView$2.executeQuery(QueryInterfaceView.java:194)
	at it.unibz.inf.ontop.protege.views.QueryInterfaceView$2.executeQuery(QueryInterfaceView.java:161)
	at it.unibz.inf.ontop.protege.gui.action.OBDADataQueryAction$1.run(OBDADataQueryAction.java:152)
```
However, the same query without the optional clause works fine and returns the rdfs:label annotations correctly.

### Steps to Reproduce

1. Recreate a database. In my example code I have a database with only one table called PERSON with the columns ID (NUMBER), NAME (VARCHAR2), SURNAME (VARCHAR2) and BIRTH (DATE).
2. Insert a person in the database. I have the following example:
![imagen](https://cloud.githubusercontent.com/assets/11376639/26109370/b11de51a-3a4f-11e7-8e95-63ec608e191e.png)

3. Create the ontology. Only a class named Person and the data properties hasBirth (domain = Person, range = xsd:date), hasId  (domain = Person, range = xsd:int), hasName  (domain = Person, range = xsd:string) and hasSurname  (domain = Person, range = xsd:string).

4. Add explicitly the rdfs:label annotation into the annotation property hierachy. If this step is ommited, you can not add the rdfs:label in the mappings file.

5. Create the mappings file with the following entry:
```
mappingId	Person
target		:person{ID} a :Person ; :hasName {NAME} ; :hasSurname {SURNAME} ; :hasBirth {BIRTH}^^xsd:date ; :hasId {ID} ; rdfs:label {NAME} , {SURNAME} . 
source		SELECT
			   ID,
			   NAME,
			   SURNAME,
			   to_char(BIRTH,'YYYY-MM-DD') AS BIRTH
			   
			FROM PERSON
```

6. Perform the query, which finishes in error:
![imagen](https://cloud.githubusercontent.com/assets/11376639/26109962/2ac0b66c-3a51-11e7-89e3-6b49af843f4a.png)

**Expected behavior:** The results must be the following (this is obtained by removing the optional clause, but the only individual has labels, so the results would be the same):
![imagen](https://cloud.githubusercontent.com/assets/11376639/26110059/7a48acd0-3a51-11e7-9290-561854bcab0c.png)

**Actual behavior:** The error appears.

**Reproduces how often:** Always

### Versions

For this test I'm using Protege 5.0.0 and Ontop 1.18.0. The database is oracle. 
However, this error is happening in an application that I'm developing with the version 1.18.1-SNAPSHOT of Ontop (develop branch in GIT).

**Comment by skomlaebri** [comment]
HI,
the error is due to a bug in ontop, that at the moment it's not able to handle the optional when there are multiple matching triples (In this case the two rdfs:label triples).
A better handling of left join is our current work and it will be added in our next release together with major refactoring.

**Comment by fanavarro** [comment]
Hi @skomlaebri, thanks for your answer.
Is there any scheduled date for the new release?

Thanks again.

Cheers,
Fran.

**Comment by skomlaebri** [comment]
Multiple matching triples are handled in version3-beta

**Comment by fanavarro** [comment]
Hi @skomlaebri . When I try to start the reasoner in version3-beta (I've downloaded the new protege bundle) with the same database and ontology exposed before it fails with the following message:

![imagen](https://user-images.githubusercontent.com/11376639/30100090-f097f438-92e7-11e7-9dc3-77ce0e8be1e0.png)

However, if I put  :hasId {ID}^^xsd:int in the mapping, instead of only  :hasId {ID}, it works. I think this behaviour is related to #179. Should we specify the datatype in the mapings in this new version in order to not to be treated as strings?

On the other hand, the query works now, so the main problem of this issue has been solved. I think we can close it.

Cheers,
Fran.
